### DISCUSSION TITLE
Allow different Layout/AlignParameters behavior with and without brackets

### DISCUSSION THREAD

**Discussion by mvz** [rubocop/rubocop#10049] [Ideas]
**Is your feature request related to a problem? Please describe.**

I'm always frustrated that the default parameter indentation logic of my editor does not match RuboCop.

I use Vim, and it formats multi-line method calls differently depending on whether brackets are present or not. So I get:

```
foo_bar baz,
  quux

foo_bar(baz,
        quux)
```

In general I like the resulting layout. However, RuboCop does not, and corrects the first version to line up `baz` and `quux`.

**Describe the solution you'd like**

I'd like another `EnforcedStyle` option for the `Layout/AlignParameters` cop that would match the behavior of Vim. I'm not sure what the name of this option would be. Perhaps `context_dependent`?

**Describe alternatives you've considered**

I've tried to configure this cop to the behavior I want, but neither of the `EnforceStyle` options does the right thing.

Disabling the cop would be throwing the baby out with the bathwater, since sometimes the alignment gets messed up and I would like this to be detected.

I could try to figure out if I can change Vim's configuration, but I rather like its default behavior.

**Comment by mvz** [comment]
I'm still interested in this.

**Comment by mvz** [comment]
This is still relevant.

**Comment by mvz** [comment]
I'm wondering if anyone else has this problem or it's just me ...

**Comment by mvz** [comment]
:wave:

**Comment by dvandersluis** [comment]
Is this a style that would make sense to have in RuboCop? /cc @rubocop/rubocop-core
